compiler/rustc_typeck/src/check/check.rs

```diff
@@ -543,10 +543,9 @@ pub(super) fn check_opaque_for_inheriting_lifetimes(
 
         if let Some(ty) = prohibit_opaque.break_value() {
             let is_async = match item.kind {
-                ItemKind::OpaqueTy(hir::OpaqueTy { origin, .. }) => match origin {
-                    hir::OpaqueTyOrigin::AsyncFn => true,
-                    _ => false,
-                },
+                ItemKind::OpaqueTy(hir::OpaqueTy { origin, .. }) => {
+                    matches!(origin, hir::OpaqueTyOrigin::AsyncFn)
+                }
                 _ => unreachable!(),
```

Comment:
Is this `unreachable` here just to satisfy the compiler or as an intentional sanity check? If it's the former you could merge the inner and the outer `match` and just do
```
let is_async = matches!(item.kind, ItemKind::OpaqueTy(hir::OpaqueTy { hir::OpaqueTyOrigin::AsyncFn, .. });
```

Reply:
I'm not sure, I'd rather leave it in case it functions as some sort of assert!()...

Reply:
Yeah, that's my concern as well. When in doubt better not to change it. Can always be done as a followup simplification.